### PR TITLE
[23] Added initial terraform deployment

### LINKED ISSUE #23: terraform deployment
To add terraform deployment for AWS

### PR DESCRIPTION
**Pull request by victorskl**
Fixed #23

### CODE REVIEW

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+1. Create AWS S3 bucket and DynamoDB table for terraform remote state and locking. Refer `backend.conf.sample` for sample backend config.
+    - Enable `Versioning` for S3 bucket
+    - Use `LockID` as Primary key, String type for DynamoDB table
+    - Then, initialize terraform with `-backend-config` flag:
+    ```
+    cp backend.conf.sample backend.conf
+    terraform init -backend-config="backend.conf"
+    ```
+    
+2. Create the following environment specific secret parameters in AWS Parameter Store. If you are planning for more than one environment, replicate more of these parameters with respective environment shorthand variables: `env`, `staging`, `prod`
```
Comment (andrewpatto) [inline terraform/README.md]: I think 'dev' was meant here rather than 'env'...

Comment (victorskl) [reply terraform/README.md]: Thanks Andrew. I have updated it.

**Comment by brainstorm** [comment]
@andrewpatto What do you think about the current state of this pullrequest? We would like to have the work of @victorskl as integrated as possible on upstream panelapp so that as much people as possible can benefit from it.

We do appreciate your feedback and we'll take it into account if we need to tweak some of the aspects of it?

/cc @ohofmann @MrsLaviniaG

**Comment by andrewpatto** [comment]
We did a brief install of the early PR - and it looked good - worked for us out of the box. The changes since then seemed even better for an out of the box experience. So its a thumbs up from us.

**Review by andrewpatto — APPROVED** [review]

**Comment by nicusX** [comment]
Thanks for the submission @victorskl. 
We are working to a cloud-native porting of panelapp at the moment. 
It will run containers in Fargate and using SQS and S3. We are also creating the Terraform code for managing it and GitLab CI/CD.
The code of the application is in the [cloud-native](https://github.com/genomicsengland/panelapp/tree/cloud-native) branch at the moment but it still needs some fixes. Terraform code and GitLab CI/CD pipeline definition will be also open-sourced, but they are not yet.
When published, it will be good to port the Cognito integration into it

**Comment by ohofmann** [comment]
@nicusX Thanks for the update - surely there must be a way to coordinate development? As you know we've been working on this for a while now to enable the panel exchange between GEL and Australian Genomics.

**Comment by brainstorm** [comment]
Indeed, @nicusX, please take a look at this PR and coordinate with @victorskl. I've noticed that some of the deployment gaps (no terraform nor cloudformation yet in your branch, cognito, for example) can be complemented with Victor's work, there's no need to have two competing solutions to solve the same problem(s).

**Comment by ohofmann** [comment]
And I should probably cc @arendon to this thread.

**Comment by nicusX** [comment]
@brainstorm The Terraform code for the full infra and deployment, along with the GitLab CI/CD, are in a different repo at the moment, not yet open-sourced. 
We are solving some integration issues specific to Genomic England then everything will be published. When finalised, integrating support for Cognito is definitely the next step

**Comment by ohofmann** [comment]
@nicusX Understood. What we're saying is that we'd like to avoid a duplication of efforts. A call / exchange of roadmaps might be helpful here so we don't end up with multiple ways of deploying PanelApp.

**Comment by victorskl** [comment]
Thanks all. Closing this PR as terraform deployment is redundant. I will create separate PR for porting Cognito integration to the `cloud-native` branch.

**Comment by gerlv** [comment]
@victorskl just wanted to let you know, we will be retiring this repo (https://github.com/genomicsengland/panelapp) and moving all development to gitlab:
https://gitlab.com/genomicsengland/panelapp
There are two repos, one for the code and another for infrastructure.
`cloud-native` has been merged into `develop` and `master` last week as we prepare for prod rollout.
I'll add a notice to this repo later today.

**Comment by victorskl** [comment]
Sure @gerlv Thanks for letting me know. I will follow up there.

**Pull request closed without merging**